rest/admin_api.go

```diff
@@ -711,19 +711,17 @@ func (h *handler) deleteUser() error {
 		}
 		return err
 	}
-	return h.db.Authenticator().Delete(user)
+	return h.db.Authenticator().DeleteUser(user)
 }
 
 func (h *handler) deleteRole() error {
 	h.assertAdminOnly()
-	role, err := h.db.Authenticator().GetRole(mux.Vars(h.rq)["name"])
-	if role == nil {
-		if err == nil {
-			err = kNotFoundError
-		}
-		return err
+	purge := h.getBoolQuery("purge")
+	found, err := h.db.DeleteRole(mux.Vars(h.rq)["name"], purge)
+	if !found {
```

Comment:
It looks like db.DeleteRole is returning found=false for any GetRole error, which results in deleteRole always returning kNotFoundError in that case.  This seems like a change from the previous handling.

In additional, it seems like we could simplify db.DeleteRole to just return error, and have it return kNotFoundError when it's not found.

Reply:
Done this and switched the 'if' ordering a little so it more closely matches the existing implementation.